mythplugins/mythmusic/mythmusic/avfdecoder.cpp

```diff
@@ -553,27 +554,32 @@ void avfDecoder::checkMetatdata(void)
 
     if (av_opt_get(m_inputContext->getContext(), "icy_metadata_packet", AV_OPT_SEARCH_CHILDREN, &pdata) >= 0)
     {
-        QString s = QString::fromUtf8((const char*) pdata);
+        QString shout = QString::fromUtf8((const char*) pdata);
 
-        if (m_lastMetadata != s)
+        if (m_lastMetadata != shout)
         {
-            m_lastMetadata = s;
-
-            LOG(VB_PLAYBACK, LOG_INFO, QString("avfDecoder: shoutcast metadata changed - %1").arg(m_lastMetadata));
-
+            m_lastMetadata = shout;
             ShoutCastMetaParser parser;
             parser.setMetaFormat(gPlayer->getDecoderHandler()->getMetadata().MetadataFormat());
+            ShoutCastMetaMap meta_map = parser.parseMeta(shout);
 
-            ShoutCastMetaMap meta_map = parser.parseMeta(m_lastMetadata);
+            QString parsed =  meta_map["title"] + "\\" + meta_map["artist"] + "\\" + meta_map["album"];
+            if (m_lastMetadataParsed != parsed)
+            {
+                m_lastMetadataParsed = parsed;
 
-            MusicMetadata mdata =  gPlayer->getDecoderHandler()->getMetadata();
-            mdata.setTitle(meta_map["title"]);
-            mdata.setArtist(meta_map["artist"]);
-            mdata.setAlbum(meta_map["album"]);
-            mdata.setLength(-1ms);
+                LOG(VB_PLAYBACK, LOG_INFO, QString("avfDecoder: shoutcast metadata changed - %1").arg(shout));
```

Comment:
You have two log messages, but only print them if both conditions (line 559 and line 567) are true.
Would it make sense to leave this message at line 562 (maybe at the LOG_DEBUG level), so that when debugging you can distinguish between "new" metadata arriving, and the new metadata parsing differently than before? Otherwise, this and the subsequent message seem redundant.

Reply:
Another argument for moving it back is that the current code prints this message before printing the two messages from parse.parseMeta.  With this change it prints after.

Reply:
Your suggestion would work for most streams.  But for a verbose one like found in this bug report, line 562 runs once per second due to the changing timestamp outside the title in the shout string.  So I can only afford to report INFO level after the parse and line 567 condition to report it only once at the title change.  I like to see the raw data and parsed result even at INFO level, as in v33.

Line 564 hands off to the parse which reports the original raw and parsed only at the DEBUG level.  This way you can see the once per second raw data in DEBUG mode for a verbose stream like this one.